### PR TITLE
Implement ExceptionArgument

### PR DESCRIPTION
**Pull request by toor1245**

### CODE REVIEW

**Comment by Dilorfin** [comment]
As temp solution it's better than was before, but I think specific exceptions will be better

**Review by Dilorfin — APPROVED** [review]

File: src/MatrixDotNet/Extensions/MatrixExtension.cs [hunk 1]
```diff
@@ -85,15 +85,12 @@ public static T[] GetColumn<T>(this Matrix<T> matrix, int index)
...
-            int y = matrix.Columns;
-            int c = x & ((x - y) >> 31) | y & (~(x - y) >> 31);
-
-            if (array.Length != c)
-                throw new MatrixDotNetException(
-                    $"Length of matrix not equal matrix Min(Rows:{matrix.Rows},Columns:{matrix.Columns})");
+            if (!matrix.IsSquare)
+            {
+                throw new MatrixNotSquareException();
+            }
```
Comment (Dilorfin) [inline src/MatrixDotNet/Extensions/MatrixExtension.cs:91]: Shouldn't we also compare array.Length with rows/columns?

**Comment by Dilorfin** [comment]
Several questions about base MatrixDotNetException class

As for me `Console.ForegroundColor` isn't something that expected to change while catching exceptions. May be it should be removed?

With all these changes, how often are used `argument2` and `argument2`?

What about best practices [[link](https://docs.microsoft.com/en-us/dotnet/standard/exceptions/how-to-create-user-defined-exceptions)]? I'm even not talking about inheriting from .net ArgumentException... Why not possible to write smth like:
```c#
try {
    ...
} catch (MatrixDotNetException e) {
    throw new MatrixDotNetException("MyMessage", e);
}
```

**Comment by toor1245** [comment]
> Several questions about base MatrixDotNetException class
> 
> As for me `Console.ForegroundColor` isn't something that expected to change while catching exceptions. May be it should be removed?
> 
> With all these changes, how often are used `argument2` and `argument2`?
> 
> What about best practices [[link](https://docs.microsoft.com/en-us/dotnet/standard/exceptions/how-to-create-user-defined-exceptions)]? I'm even not talking about inheriting from .net ArgumentException... Why not possible to write smth like:
> 
> ```cs
> try {
>     ...
> } catch (MatrixDotNetException e) {
>     throw new MatrixDotNetException("MyMessage", e);
> }
> ```

As I remember, argument1 and argument used a long time ago, so we can easily delete this properties.

**Review by Dilorfin — APPROVED** [review]

**Pull request merged**